### DISCUSSION TITLE
Is there a way of making dynamic attributes to be converted to strings automatically with JSON.stringify?

### DISCUSSION THREAD

**Discussion by ilse-langnar** [alpinejs/alpine#3650] [1. Help]
I have the following setup:

```html
<div>
  <div :json='{ "property": "value" }' > </div>
<div>
```
I need ":json" to be automatically converted to a string using "JSON.stringify()" and not to "[object Object]". I'm trying to implement a "component system" for alpine and behavior is "destroying" the props I pass, I want to pass "complex" data structures such as objects and arrays, if they're converted on the fly to string using "JSON.stringify" I don't care. I just can't have my json become "[object object]" because the operation is irreversible. 

I'm currently doing this:

```html
<div :json='JSON.stringify( { "property": "Value" } )' > </div> 
```

and it works but overall I would like to know if there is a way of passing objects to either "x-bind" or ":" and have them not dissapear into "[object object]". Thank you in advance.

**Comment by ekwoka** [comment]
You could do three things:

patch `Object.prototype.toString` to use `JSON.stringify` (not recommended) and will probably break some stuff

You can make a custom directive like `x-stringify` where you could do `x-stringify:json` to have it stringify the value that is bound instead.

You could potentially make a PR to add a modifier to `x-bind` so have like `x-bind.stringify` or something (so `:json.stringify` in your examples shorthand).
